src/simulation/m_time_steppers.fpp

```diff
@@ -980,6 +1004,187 @@ contains
+        call s_compute_el_coupled_solver(q_cons_ts(2)%vf, q_prim_vf, rhs_ts_adapt(4)%vf, RKstep)
+        call s_update_tmp_rkck(RKstep, q_cons_ts, rhs_ts_adapt, q_prim_vf)
+        if (lag_largestep) goto 502
+
+        ! Fifth time-stage
+        time_tmp = time_prev + 1.0d0*dt
+        RKstep = 5
+!$acc update device (time_tmp)
+#ifdef DEBUG
+        if (proc_rank == 0) print *, 'RKCK 5th time-stage at', time_tmp
+#endif
+        call s_compute_rhs(q_cons_ts(2)%vf, q_prim_vf, rhs_ts_adapt(5)%vf, pb_ts(1)%sf, rhs_pb, mv_ts(1)%sf, rhs_mv, t_step, time_avg)
+        call s_compute_el_coupled_solver(q_cons_ts(2)%vf, q_prim_vf, rhs_ts_adapt(5)%vf, 5)
+        call s_update_tmp_rkck(5, q_cons_ts, rhs_ts_adapt, q_prim_vf)
+        if (lag_largestep) goto 502
+
+        ! Sixth time-stage
+        time_tmp = time_prev + 0.875d0*dt
+        RKstep = 6
+!$acc update device (time_tmp)
+#ifdef DEBUG
+        if (proc_rank == 0) print *, 'RKCK 6th time-stage at', time_tmp
+#endif
+        call s_compute_rhs(q_cons_ts(2)%vf, q_prim_vf, rhs_ts_adapt(6)%vf, pb_ts(1)%sf, rhs_pb, mv_ts(1)%sf, rhs_mv, t_step, time_avg)
+        call s_compute_el_coupled_solver(q_cons_ts(2)%vf, q_prim_vf, rhs_ts_adapt(6)%vf, 6)
+        call s_update_tmp_rkck(6, q_cons_ts, rhs_ts_adapt, q_prim_vf)
+        if (lag_largestep) goto 502
+
+        if (cfl_dt) then
+            ! Truncation Error
+#ifdef DEBUG
+            if (proc_rank == 0) print *, 'Computing truncation error (4th/5th RKCK)'
+#endif
+            call s_calculate_rkck_truncation_error()
+        end if
+
+502     if (lag_largestep) then ! Encountered negative radius, so reduce dt and restart time step
+            if (cfl_dt) then
+                if (RKh > 1.0d-14) then
+                    RKh = RKh/2.0d0
+                    if (proc_rank == 0) print *, '>>>>> WARNING: Reducing dt and restarting time step, now dt: ', RKh
+                    lag_largestep = .false.
+                    dt = RKh
+                    !$acc update device(lag_largestep, dt)
+                    goto 501
+                else
+                    call s_mpi_abort('Time step smaller than 1e-14')
+                end if
+            else
+                call s_mpi_abort('Time step too large, please reduce dt or enable cfl_adapt_dt')
+            end if
+        end if
+
+        if (cfl_dt) then ! Checking truncation error
+            lag_errmax = min(lag_errmax, 1.0d0)
+            if (num_procs > 1) then
+                call s_mpi_allreduce_max(lag_errmax, errmax_glb)
+                lag_errmax = errmax_glb
+            end if
+            lag_errmax = lag_errmax/lag_rkck_tolerance ! Scale relative to user required tolerance.
+
+            if ((lag_errmax > 1.0d0)) then   ! Truncation error too large, reduce dt and restart time step
+                htemp = SAFETY*RKh*((floor(lag_errmax*1.0d05)/1.0d05)**PSHRNK)
```

Comment:
where does 1e5/1e5 come from? either way, if we need something like this it seems like a magic number that should go to `m_constants`

Reply:
I found the neceessity to round the relative truncation error (5th decimal) to avoid the inclusion of random decimals after dividing by a very small number (rkck_tolerance). Fixed.